src/r2mm/profiles/ProfilesClient.ts

```diff
@@ -19,6 +19,19 @@ function formatApiError<T>(e: T, genericTitle: string): R2Error | T {
                 message,
                 "You were rate limited by the server, wait for a while and try again."
             );
+        } else if (e.response.status === 404) {
+            // Note that this error text is only applicable when importing, but it should be OK as creating a profile shouldn't respond with 404 anyway.
+            return new R2Error(
+                genericTitle,
+                `Server responded with \"404: Not Found\" while trying to download the profile`,
+                `The profile import code entered might either be expired or contain typos.`
+            );
+        } else if (e.response.status.toPrecision(1) === "5e+2") { // Status code is 5xx
```

Comment:
- CDN issues do not return 500 error code, or any error code for that matter, since the destination server is not reached at all. They manifest as `Network Error`s
- These are moot points due to the above, but I wrote them before the first one so FYI:
    - While what's done hear is neat (and a TIL moment for me), I think the code isn't that readable without the comment accompanying it. I would prefer `(e.response.status > 499 && e.response.status < 600)` for readability's sake.
    - The error message says "while trying to download", but I assume uploads might return 500 errors as well